community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/IndexingService.java

```diff
@@ -308,9 +314,65 @@ public void accept( Long indexId, IndexProxy proxy )
         samplingController.recoverIndexSamples();
         samplingController.start();
 
+        // So at this point we've started population of indexes that needs to be rebuilt in the background.
+        // Indexes backing uniqueness constraints are normally built within the transaction creating the constraint
+        // and so we shouldn't leave such indexes in a populating state after recovery.
+        // This is why we now go and wait for those indexes to be fully populated.
```

Comment:
I think this new behaviour means that we will wait for indexes that were created for a constraint that never got committed. Such indexes will get dropped later in the recovery process. That is a bit wasteful, although if the alternative is that we risk starting the database up with a constraint in a non-operational state, this waste is much preferred.

I wonder if we could achieve the same effect by passing `true` to `proxySetup.createPopulatingIndexProxy(...)` instead? (where there is a comment about `false` being "surprising")